### ISSUE TITLE
Video Block - Video fails to upload on self-hosted Jetpack site.

### ISSUE THREAD

**Issue by ttahmouch** [wordpress-mobile/gutenberg-mobile#4503]
**Describe the bug**
There appears to be issues if you are using a self-hosted site. I tested with a generated, self-hosted Jetpack site. The upload UX appeared to work and would show the file uploading and the `Video` block changing metadata in the block editor HTML mode. It even showed the block edits in the `wp-admin` page as well, but the video files weren't able to be found in the `Media` section of `wp-admin`; only image files from previous `Image` block tests were found. When I switched to an `Atomic` site everything worked as expected, i.e., when the file finished upload I saw an image thumbnail of the first frame of the video as opposed to an error that said "Media Options > File Type Unknown."

[Sanity Tests](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4438#issuecomment-1016634668)

**To Reproduce**
Video block - Insert video from device (cancel) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/video.md#tc002)

**Expected behavior**
Video uploads should be available on the self-hosted Jetpack site in the Media section, and the path to the video file should be a URL to the file from the Media section; not a local file path from the phone OS.

**Smartphone (please complete the following information):**
iPhone 13 Pro Max (15.1.1)

**Additional context:**
Sanity Test - 1.69.0

**Screenshots**
**Video URI is Local; Not Remote:**
![image](https://user-images.githubusercontent.com/3058263/151205073-424e29c0-4486-439b-9a81-5dbe01e72c81.png)
**Video File Type Unknown when Uploaded to Jetpack Site:**
![image](https://user-images.githubusercontent.com/3058263/151205109-2e769694-44fa-4b5e-b739-345b1642add0.png)

**Uploading a video to self-hosted connected with Jetpack:**

https://user-images.githubusercontent.com/14905380/151210398-25bcbdc4-a9f6-4e9d-93af-cde61ec77737.mp4

**Comment by hypest** [comment]
Tried out WPiOS `v19.1.02` on a selfhosted Jetpack site of mine on Pressable, logged in via username/password (that is, not via WoredPress.com) and video upload seems to work OK ([post here](https://hypestpressable-jetpack.mystagingwebsite.com/2022/02/v-test-ios/)). I wonder if this has something to do with the Jurassic Ninja instance, or some weird state of the app.

**Comment by ttahmouch** [comment]
> Tried out WPiOS `v19.1.02` on a selfhosted Jetpack site of mine on Pressable, logged in via username/password (that is, not via WoredPress.com) and video upload seems to work OK ([post here](https://hypestpressable-jetpack.mystagingwebsite.com/2022/02/v-test-ios/)). I wonder if this has something to do with the Jurassic Ninja instance, or some weird state of the app.

I'm good with closing this one as well if others can't reproduce it. I may be entirely related to Jurassis Ninja instances, but more importantly if you can't replicate it on a "real" self-hosted Pressable instance, then that gives me some confidence that this is just a problem with the testing environment.

**Comment by hypest** [comment]
Mind if you give it a try on Pressable from your side? Just want to make sure this is confirmed before we close the ticket, considering it's a rather important flow (video uploads). Thanks!

**Comment by hypest** [comment]
I also tried out a new Jurassic Ninja site now and video uploads OK both from the WPAndroid (v19.3-rc-1) and WPiOS (v19.3.0.3) app. Also, I tried stopping the video upload while going and the post didn't end up containing any link to a video file when published. While at it, I verified that the Media Library in wp-admin does indeed contain the video files that finished uploading.

I'll go ahead and close this ticket at this stage, but happy to reopen if we get more info on how to reproduce the issue. Thanks!
